File: Java/Medium/TimeLimitExceeded/prison-cells-after-n-days_1.java
class Solution {
    public int[] prisonAfterNDays(int[] cells, int n) {
        
        int[] next = new int[cells.length];
        for (int i = 0; i < n; i++) {
            for (int j = 1; j < cells.length - 1; j++) {
                if (cells[j - 1] == cells[j + 1]) {
                    next[j] = 1;
                } else {
                    next[j] = 0;
                }
            }
            cells = next;
            next = new int[cells.length];
        }
        return cells;
    }
}

File: Rust/Medium/Accepted/prison-cells-after-n-days_1.rs
impl Solution {
    pub fn prison_after_n_days(cells: Vec<i32>, n: i32) -> Vec<i32> {
        
        let mut cells = cells;
        let mut n = n;
        let mut seen = vec![];
        while n > 0 {
            let mut next = vec![0; 8];
            for i in 1..7 {
                next[i] = if cells[i - 1] == cells[i + 1] { 1 } else { 0 };
            }
            if seen.contains(&next) {
                n %= seen.len() as i32;
            } else {
                seen.push(next.clone());
            }
            if n > 0 {
                n -= 1;
                cells = next;
            }
        }
        cells
    }
}